packages/react/src/primitives/types/autocomplete.ts

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react';
+
+import { SearchFieldProps } from './searchField';
+import { ViewProps } from './view';
+
+export interface AutocompleteComboboxProps {
+  role: React.AriaRole;
+  'aria-activedescendant': React.AriaAttributes['aria-activedescendant'];
+  'aria-autocomplete': React.AriaAttributes['aria-autocomplete'];
+  'aria-controls': React.AriaAttributes['aria-controls'];
+  'aria-expanded': React.AriaAttributes['aria-expanded'];
+  'aria-haspopup': React.AriaAttributes['aria-haspopup'];
+  'aria-owns': React.AriaAttributes['aria-owns'];
+}
+
+export interface Option {
+  /**
+   * @description
+   * Unique id for an option
+   */
+  id: string;
+
+  /**
+   * @description
+   * Label for an option
+   */
+  label: string;
+
+  [key: string]: string;
+}
+
+export interface AutocompleteOptionProps extends ViewProps {
+  /**
+   * React node that will be wrapped inside a li element
+   */
+  children: React.ReactNode;
+  /**
+   * @description
+   * Determine if it is an active option
+   */
+  isActive: boolean;
+}
+
+export interface AutocompleteMenuProps extends ViewProps {
+  /**
+   * @description
+   * Defines a string value that labels an interactive element for accessibility
+   */
+  ariaLabel?: React.AriaAttributes['aria-label'];
+
+  /**
+   * @description
+   * This is a slot to label a list of options
+   */
+  Header?: React.ReactNode;
```

Comment:
Missed this in my earlier review, will I be able to pass in a `Header` component that is wrapped with `forwardRef` with `ReactNode` as the type here? Same question for the other slot components.

Reply:
Yes, the example on docs have demonstrated that. `View` is wrapped with` forwardRef`

```
export const AutocompleteCustomHeaderExample = () => (
  <Autocomplete
    label="Autocomplete custom header example"
    options={options}
    menuSlots={{
      Header: (
        <View padding="xxxs">
          <View>Search results:</View>
          <Divider />
        </View>
      ),
    }}
  />
);
```